### ISSUE TITLE
Unable to add global "wordsmith" as the runtime or the extensions have already been initialized

### ISSUE THREAD

**Issue by michaelrog** [TopShelfCraft/Wordsmith#6]
Error upon installation...

LogicException: Unable to add global "wordsmith" as the runtime or the extensions have already been initialized.

Craft CMS 3.0.15
PHP 7.0.30

**Comment by arifje** [comment]
Any plans on fixing? This is currently the only plugin for truncating text... cheers!

**Comment by michaelrog** [comment]
I'm currently unable to reproduce this issue using Wordsmith 3.0.2 on Craft 3.0.16.1.

@skoften / @webmarks — Could yall make sure you're fully updated?

Assuming so, could you let us know what other plugins you have installed, and the order in which they appear in the plugins table in the db?

**Comment by arifje** [comment]
Sure, I got two installs on the same server that generate this error. 

Site 1:

```
mysql> select id, handle, version from plugins;
+----+-------------------------------+---------------+
| id | handle                        | version       |
+----+-------------------------------+---------------+
| 11 | mobile-detect                 | 1.0.1         |
| 12 | similar                       | 1.0.2         |
| 13 | preview-button                | 1.0.2         |
| 14 | transcoder                    | 1.1.3         |
| 15 | logs                          | 3.0.0         |
| 17 | image-resizer                 | 2.0.2         |
| 18 | disqus                        | 1.0.7         |
| 19 | redactor                      | 2.1.2         |
| 20 | snitch                        | 2.0.2         |
| 21 | redirect                      | 1.0.17        |
| 22 | simplemap                     | 3.3.3         |
| 23 | video-embedder                | 1.0.9         |
| 24 | social                        | 2.0.0-beta.10 |
| 26 | cp-clearcache                 | 1.0.2         |
| 27 | campaign                      | 1.0.0-beta2   |
| 28 | telegram-notification-channel | 1.0.1         |
| 32 | content-stats                 | 2.1.0         |
| 37 | admin-bar                     | 3.1.5         |
| 38 | ansel                         | 2.0.0         |
| 40 | notifications                 | 1.1.1         |
| 41 | cp-element-count              | 1.0.1         |
| 42 | delete-account                | 1.0.1         |
| 43 | locate                        | 2.0.1         |
| 45 | embeddedassets                | 1.0.2         |
| 46 | gtmetrix                      | 1.1.4         |
| 47 | upvote                        | 2.0.3         |
| 48 | entry-count                   | 2.0.1         |
| 49 | uploadit                      | 1.0.2         |
| 50 | guest-entries                 | 2.1.3         |
| 51 | http2-server-push             | 1.0.0         |
| 52 | wheelform                     | 1.9.0         |
| 53 | seomatic                      | 3.1.3         |
| 54 | sidebar-themes                | 1.0.5         |
+----+-------------------------------+---------------+
```

Site 2:

```
+----+-------------------------------+---------------+
| id | handle                        | version       |
+----+-------------------------------+---------------+
| 11 | mobile-detect                 | 1.0.1         |
| 12 | similar                       | 1.0.2         |
| 13 | preview-button                | 1.0.2         |
| 14 | transcoder                    | 1.1.3         |
| 15 | logs                          | 3.0.0         |
| 17 | image-resizer                 | 2.0.2         |
| 18 | disqus                        | 1.0.7         |
| 19 | redactor                      | 2.1.2         |
| 20 | snitch                        | 2.0.2         |
| 21 | redirect                      | 1.0.17        |
| 22 | simplemap                     | 3.3.3         |
| 23 | video-embedder                | 1.0.9         |
| 24 | social                        | 2.0.0-beta.10 |
| 26 | cp-clearcache                 | 1.0.2         |
| 27 | campaign                      | 1.0.0-beta2   |
| 28 | telegram-notification-channel | 1.0.1         |
| 32 | content-stats                 | 2.1.0         |
| 37 | admin-bar                     | 3.1.5         |
| 38 | ansel                         | 2.0.0         |
| 40 | notifications                 | 1.1.1         |
| 41 | cp-element-count              | 1.0.1         |
| 42 | delete-account                | 1.0.1         |
| 43 | locate                        | 2.0.1         |
| 45 | embeddedassets                | 1.0.2         |
| 46 | gtmetrix                      | 1.1.4         |
| 47 | upvote                        | 2.0.3         |
| 48 | entry-count                   | 2.0.1         |
| 49 | uploadit                      | 1.0.2         |
| 50 | guest-entries                 | 2.1.3         |
| 51 | http2-server-push             | 1.0.0         |
| 52 | wheelform                     | 1.9.0         |
| 53 | seomatic                      | 3.1.3         |
| 54 | sidebar-themes                | 1.0.5         |
+----+-------------------------------+---------------+

```

Server info;

PHP 7.0.30-0ubuntu0.16.04.1 (cli) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.30-0ubuntu0.16.04.1, Copyright (c) 1999-2017, by Zend Technologies

Linux testmachine 4.4.0-127-generic #153-Ubuntu SMP Sat May 19 10:58:46 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

Both run Craft CMS 3.0.16.1

**Comment by frankharrison-wj** [comment]
I've also just hit this bug. I had Wordsmith installed and working fine, but then had to delete my vendor directory and rebuild from composer. I managed to reinstall everything apart from Wordsmith which gives the same error as this thread's title.

I'm on Craft CMS 3.0.17.1, running on PHP 7.2.1. Attached is a screenshot of my plugins table...

<img width="326" alt="screen shot 2018-07-26 at 10 59 42" src="https://user-images.githubusercontent.com/9253517/43255949-182ab1aa-90c3-11e8-8379-a865d502d8b1.png">

**Comment by michaelrog** [comment]
Fixed by b198c3d. 

(c.f. https://github.com/pixelandtonic/Wordsmith/commit/4c37dc0ae7a439141f1cbb67e80555580f0cc6a7)
